packages/rfw/lib/src/flutter/runtime.dart

```diff
@@ -1095,6 +1437,17 @@ class _WidgetState extends State<_Widget> implements DataSource {
     return subscription.value;
   }
 
+  Object _widgetBuilderArgResolver(List<Object> rawDataKey) {
+    final _Key widgetBuilderArgKey = _Key(_kWidgetBuilderArgSection, rawDataKey);
+    final _Subscription subscription = _subscriptions[widgetBuilderArgKey] ??= _Subscription(
+      widget.widgetBuilderScope,
```

Comment:
i don't understand how this works. how do you get the right widget builder argument, e.g. in a nested case?

Reply:
This might help https://github.com/flutter/packages/blob/e7a48123e5e14b316a73acbd599bc9056cf4799a/packages/rfw/lib/src/flutter/runtime.dart#L548

Reply:
Not for this PR but I think we can generalize (and simplify) state, data and this newly introduced builder scope concept in a single scope entity (pretty much like interpreters).